### ISSUE TITLE
REST client 2.5.4 logs username and password on the console [Bug]

### ISSUE THREAD

**Issue by dajtxx** [thingsboard/thingsboard#4932]
There is some web request logging enabled by default and it shows the users password when logging in to Thingsboard. I have replaced the real values with "zzz" here.

The default log settings should not have the debug messages getting emitted anyway, but even with that switched off obviously the password should not be shown in cleartext.

```
14:12:54.279 [main] DEBUG org.springframework.web.client.RestTemplate - HTTP POST https://zzz/api/auth/login
14:12:54.301 [main] DEBUG org.springframework.web.client.RestTemplate - Accept=[application/json, application/*+json]
14:12:54.314 [main] DEBUG org.springframework.web.client.RestTemplate - Writing [{password=zzz, username=zzz@zzz}] with org.springframework.http.converter.json.MappingJackson2HttpMessageConverter
14:12:54.671 [main] DEBUG org.springframework.web.client.RestTemplate - Response 200 OK
```

**Comment by dajtxx** [comment]
This happens in 3.2.2 as well.

The work-around is to create a logback.xml file and put it in the classpath - in my case in the root of my application jar.

```
<configuration>
  <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
    <encoder>
      <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n</pattern>
    </encoder>
  </appender>
  <root level="INFO">
    <appender-ref ref="STDOUT" />
  </root>
</configuration>
```

This suppresses DEBUG messages.

The fact these messages are being logged by default when I include rest-client in my pom.xml and call it is a bug. I don't use any logging framework in my app so should not need to configure one.

**Comment by agtokty** [comment]
Hi @dajtxx , actually there is a[ logback.xml ](https://github.com/thingsboard/thingsboard/blob/master/application/src/main/resources/logback.xml)file already in application resources ant it used by application. And I couldn't repeat your case with default configuration.

**Comment by dajtxx** [comment]
Hi,

I have written a Java app that uses the ThingsBoard rest-client. I bring it in via the pom.xml file. It does not run on the ThingsBoard server box, it runs a on user’s laptop. I call the rest-client like this:

            rc = new RestClient("http://" + host);
            rc.login(user, password);
            final List<EntitySubtype> deviceTypes = rc.getDeviceTypes();
            rc.logout();

and I get the debug messages. This happens from either within Eclipse or from the command line when I create an executable jar for my app whose MANIFEST.MF points to the jars in a lib directory.

Where is the ThingsBoard logback.xml file in this scenario? In one of your jars? I think the only Thingsboard jars referenced in this setup are rest-client & data. Obv there are a ton of your dependencies too.

To get the messages to go away I had to create that file in my jar and set the root level to INFO.

This happened with both v2.4 and v3.2.2 of the rest-client.

Regards, David.

Here’s the output:

16:33:46.841 [main] WARN org.springframework.http.converter.json.Jackson2ObjectMapperBuilder - For Jackson Kotlin classes support please add "com.fasterxml.jackson.module:jackson-module-kotlin" to the classpath
16:33:46.879 [main] DEBUG org.springframework.web.client.RestTemplate - HTTP POST http://thingsboard-staging.farmdecisiontech.net.au:8080/api/auth/login
16:33:46.887 [main] DEBUG org.springframework.web.client.RestTemplate - Accept=[application/json, application/*+json]
16:33:46.898 [main] DEBUG org.springframework.web.client.RestTemplate - Writing [{password=wascleartext, ***@***.***}] with org.springframework.http.converter.json.MappingJackson2HttpMessageConverter
16:33:47.155 [main] DEBUG org.springframework.web.client.RestTemplate - Response 200 OK
16:33:47.160 [main] DEBUG org.springframework.web.client.RestTemplate - Reading to [com.fasterxml.jackson.databind.JsonNode]
16:33:47.171 [main] DEBUG org.springframework.web.client.RestTemplate - HTTP GET http://thingsboard-staging.farmdecisiontech.net.au:8080/api/device/types
16:33:47.175 [main] DEBUG org.springframework.web.client.RestTemplate - Accept=[application/json, application/*+json]
16:33:47.201 [main] DEBUG org.springframework.web.client.RestTemplate - Response 200 OK
16:33:47.204 [main] DEBUG org.springframework.web.client.RestTemplate - Reading to [java.util.List<org.thingsboard.server.common.data.EntitySubtype>]
16:33:47.253 [main] DEBUG org.springframework.web.client.RestTemplate - HTTP POST http://thingsboard-staging.farmdecisiontech.net.au:8080/api/auth/logout
16:33:47.310 [main] DEBUG org.springframework.web.client.RestTemplate - Response 200 OK

**Comment by ashvayka** [comment]
Hi @dajtxx , we have fixed this in master and the fix will be available since 3.3.0
